source/Public/Get-SqlDscStartupParameter.ps1

```diff
@@ -65,19 +65,19 @@ function Get-SqlDscStartupParameter
         $InstanceName = 'MSSQLSERVER'
     )
 
-    $previousErrorActionPreference = $ErrorActionPreference
+    $originalErrorActionPreference = $ErrorActionPreference
 
     $ErrorActionPreference = 'Stop'
 
     Assert-ElevatedUser -ErrorAction 'Stop'
 
+    $ErrorActionPreference = $originalErrorActionPreference
+
     if ($PSCmdlet.ParameterSetName -eq 'ByServiceObject')
     {
         $ServiceObject | Assert-ManagedServiceType -ServiceType 'DatabaseEngine'
     }
```

Comment:
This line should probably move below Assert-ManageServiceType. @coderabbitai can you create an issue to tracke this?